Fixed the channel selection P300

brianjohannes opened the pull request:
The channel subsets chosen by the channel selection method weren't being used for online classification.

This also simplifies the calculation of posterior probabilities from epochs.

This was tested with p300_offline_test.py and the debugger to ensure subsets were being used.

ekinney-lang reviewed (changes requested):
I cannot get this to run online without running into an error that is appearing to be part of the main.py branch.
Steps to reproduce:
1. Open up BCI Essentials Python in terminal
2. Open a new eeg_lsl_sim.py stream (pointing toward example_p300 data).
3. Open BCI Essentials Unity, and populate scene with pre-made set-up matrix.
4. Hit "T" to start training. Immediately the BCI-Essentials Python terminal is broken.

![image](https://github.com/user-attachments/assets/1273d069-5569-40d4-a61a-d7fcd4e46941)

brianjohannes commented:
I think it might be fixed now. In p300_paradigm, I changed the marker index that it was getting the train target from to be 0 instead of 3 (idk why it was 3). @ekinney-lang could you give this a try. I also squashed another bug in the process, which was that declared subsets weren't being used by the classifier. You can test this by passing a subset into ErpRgClassifier.

brianjohannes commented:
Hey @ekinney-lang I changed prediction labels to be integers. Does this solve your issue? I have been unable to replicate the bug with seven objects. For context, I used 7 objects, 6 total selections, and 2 flashes per object in my testing. Let me know how it goes!

ekinney-lang reviewed (approved):
These look like they will be fine, but am still getting the same error with the sklearn package in newer version of Unity. Will try to hunt it down.

ekinney-lang commented:
NOTE: In Bessy-Unity v1.1.8 the "Single Flash" option remains broken with Bessy-python, while the "Context Aware" single flash does not. Unsure why this is happening.
